### PR TITLE
MHV-48126 Call PHR refresh asynchronously after MHV user login

### PR DESCRIPTION
**Pull request by mmoyer-va**
## Summary
- After a user logs in to VA.gov, check if they are an MHV user and then call the PHR Refresh endpoint.
- This is done in a Sidekiq job so as not to slow down the login
- The Sidekiq job will only be enqueued for MHV users
- This action is performed on login because waiting until the user makes a Medical Records request is too late. The refresh takes time to run, so putting it as early in the process as possible is desired for a smooth user experience.
- Added a feature flag so the refresh can be easily enabled/disabled in various environments

### [MHV-48126](https://jira.devops.va.gov/browse/MHV-48126) - Implement PHR refresh timing and when to call it ###

> Understand PHR refresh timing and when to call it
> 
> User Story: As a Medical Records Developer, I want to Understand PHR refresh timing and when to call it, so that I can develop code using the PHR refresh functionality.
> 
> Acceptance Criteria:
> AC1 Understand PHR Refresh functionality
> AC2 Understand when to call the PHR refresh for MR domains
> AC3
> AC4
> AC5
> 
> Links to UCD: N/A
> 
> Architectural Notes:

## Testing done
- Manually tested that the job could be created and run successfully via the rails command line
- Rspec test for the job, ensuring that its logic is correct, including error cases
- Rspec test for AfterLoginActions, ensuring that the job is added to the Sidekiq queue (only if feature flag enabled)

## What areas of the site does it impact?
- This code was created for Medical Records, but the Sidekiq job will be run--and the PHR refresh called--**for every MHV user** upon login. Any errors are logged.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by tynyttie — APPROVED** [review]
I approve these changes.

**Review by LindseySaari — APPROVED** [review]

File: app/services/login/after_login_actions.rb [hunk 1]
```diff
@@ -28,6 +28,12 @@ def perform
         TestUserDashboard::UpdateUser.new(current_user).call(Time.current)
         TestUserDashboard::AccountMetrics.new(current_user).checkout
       end
+
+      if Flipper.enabled?(:mhv_medical_records_phr_refresh_on_login,
+                          @current_user) && current_user.mhv_correlation_id.present?
+        MHV::PhrUpdateJob.perform_async(current_user.icn,
+                                        current_user.mhv_correlation_id)
```
Comment (bosawt) [inline app/services/login/after_login_actions.rb:35]: I don't think you need to gate this on `mhv_correlation_id` if you're also checking it during the job itself

**Review by bosawt — CHANGES_REQUESTED** [review]
Mainly just making sure `mhv_correlation_id` is what you actually want

**Comment by bosawt** [comment]
Also, I tested this locally and on prod and it doesn't seem to work in either case? At least the refresh functionality itself:

image

**Comment by bosawt** [comment]
We need to be extremely careful about the functionality that's added to auth, I would make sure this refresh is actually working through standard flows before introducing it to the authentication flow

File: app/workers/mhv/phr_update_job.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    private
+
+    def run_refresh(icn)
+      phr_client = PHRMgr::Client.new
+      phr_client.post_phrmgr_refresh(icn)
+    end
+
+    def mhv_user?(mhv_correlation_id)
+      mhv_correlation_id.present?
```
[the remark this answers is not in the document]
Comment (mmoyer-va) [reply app/workers/mhv/phr_update_job.rb:30]: Sure -- I only want to run this refresh for users who have an MHV account. I figured that the correlation ID will be present for all such users but if there is a better way I can make the switch!

**Comment by mmoyer-va** [comment]
@bosawt You're right -- we haven't added the `x_auth_key` in production yet. It does work locally, but that param must be added to the settings.local.yml file.

However I don't think we want to run this in production before we enter Phase 0. We should turn off that feature flag in prod until that point.

**Comment by bosawt** [comment]
> @bosawt You're right -- we haven't added the `x_auth_key` in production yet. It does work locally, but that param must be added to the settings.local.yml file.
> 
> However I don't think we want to run this in production before we enter Phase 0. We should turn off that feature flag in prod until that point.

Ah, I see, thanks for that context. So feature flags should be used to perform rollouts of code that's expected to work, not to hide code that doesn't currently work, especially in flows that are currently live (and heavily trafficked like this). I think we should revisit the concept of adding it here once this refresh is working and you are at a phase where you'd like it to be added.

**Comment by mmoyer-va** [comment]
@bosawt I'd like to revisit this PR. The `x_auth_key` has been added to the parameter store for production, and the SideKiq job has been tested in staging (currently being run when a user accesses the MR page). Let me know what other concerns you have so we can discuss, as there is a strong desire to move this action left as far as possible. Otherwise users may see an empty list when in fact they have records.

**Comment by ericboehs** [comment]
Moving to draft to remove from review queue.

**Pull request closed without merging**